algoliasearch/src/main/java/com/algolia/search/saas/AbstractClient.java

```diff
@@ -275,12 +296,30 @@ public int getSearchTimeout() {
      * @param searchTimeout The read timeout for search requests (ms).
      */
     public void setSearchTimeout(int searchTimeout) {
-        if (searchTimeout <= 0)
-            throw  new IllegalArgumentException();
+        checkTimeout(searchTimeout);
         this.searchTimeout = searchTimeout;
     }
 
     /**
+     * Get the timeout for retrying connection to a down host.
+     *
+     * @return The delay before connecting again to a down host (ms).
+     */
+    public int getHostDownDelay() {
```

Comment:
I like the "host down delay" name, better than the one I used in Swift ("host status timeout"). The problem is the Swift version is already released... :/ Alright, just keep it, I will maybe one day introduce a new property and deprecate the old one.